tests/integration/ha_tests/helpers.py

```diff
@@ -334,3 +341,109 @@ async def get_units_hostnames(ops_test: OpsTest) -> List[str]:
         f"{unit.name.replace('/', '-')}.mongodb-k8s-endpoints"
         for unit in ops_test.model.applications[APP_NAME].units
     ]
+
+
+async def db_step_down(ops_test: OpsTest, sigterm_time: datetime) -> None:
+    """Pipes the k8s logs, looking for stepdown message."""
+    kubectl_cmd = [
+        "microk8s",
+        "kubectl",
+        "logs",
+        f"-n{ops_test.model_name}",
+        f"--since-time={sigterm_time.isoformat()}",
+        "",
+        "-c",
+        MONGODB_CONTAINER_NAME,
+        "-f",
+    ]
+
+    grep_cmd = (
+        "grep",
+        "-m1",
+        '"Starting an election due to step up request"',
+    )
+
+    procs = []
+    for unit in ops_test.model.applications[APP_NAME].units:
+        kubectl_cmd[5] = unit.name.replace("/", "-")
+        kubectl_proc = subprocess.Popen(kubectl_cmd, stdout=subprocess.PIPE)
+        grep_proc = subprocess.Popen(
+            grep_cmd, stdin=kubectl_proc.stdout, stdout=subprocess.DEVNULL
```

Comment:
The election log may drift away, due to `continuous_writes()` spam, so I pipe the logs to make sure it's not missed. Is this a sane approach?

Reply:
YES! Awesome work here, this was a pain for me to deal with as well.